controllers/mlmd.go

```diff
@@ -57,7 +60,24 @@ func (r *DSPAReconciler) ReconcileMLMD(dsp *dspav1alpha1.DataSciencePipelinesApp
 			return err
 		}
 	} else {
-		err := r.ApplyDir(dsp, params, mlmdTemplatesDir)
+		err := r.ApplyDir(dsp, params, mlmdTemplatesDir+"/"+mlmdGrpcService)
+		if err != nil {
+			return err
+		}
+
+		if params.PodToPodTLS {
+			var certificatesExist bool
+			certificatesExist, err = params.LoadMlmdCertificates(ctx, r.Client, r.Log)
+			if err != nil {
+				return err
+			}
+
+			if !certificatesExist {
+				return errors.New("secret containing the certificate for MLMD gRPC Server was not created yet")
```

Comment:
I'm not an expert on this, but I would expect to see a more of a warning and a re-reconcile, not an error. I think the error might look a little more scary in the logs as well. I'd like Humair's opinion though.

Reply:
I agree. But an error is necessary to update the DSPA status with `not ready`.
Maybe we should improve the DSPA status mechanism to handle these situations.

@HumairAK thoughts?

Reply:
yeah we don't have a protocol for handling readying state, this will have to do for now

I think the solution is relatively simple, just have the independent component reconcilers functions return a status condition/message, and put that for ready status, but this can be it's own separate Jira, @hbelmiro can you create one?

Reply:
https://issues.redhat.com/browse/RHOAIENG-13321